src/Reflection/ClassReflection.php

```diff
@@ -776,22 +776,19 @@ public function getTypeAliases(): array
 			$typeAliasImportTags = $resolvedPhpDoc->getTypeAliasImportTags();
 			$typeAliasTags = $resolvedPhpDoc->getTypeAliasTags();
 
+			$nameScope = $resolvedPhpDoc->getNullableNameScope() ?? new NameScope(null, []);
```

Comment:
Is this kind of thing ok? The ResolvedPhpDocBlock internally uses `$nameScope` as non-nullable, but publicly exposes it as nullable. In what circumstances can it actually be null?

Reply:
I posted what could be done instead as comments :) I guess name scope is null in these situations: https://github.com/phpstan/phpstan-src/blob/927d442c8cd0339f3beb12ff18971fa2b3328ba5/src/Type/FileTypeMapper.php#L105-L111

And also: https://github.com/phpstan/phpstan-src/blob/927d442c8cd0339f3beb12ff18971fa2b3328ba5/src/PhpDoc/ResolvedPhpDocBlock.php#L165

Basically when it knows that NameScope isn't going to be needed (because all tags are already resolved) it doesn't assign it.

Reply:
Thanks for the explanation! I've fixed that according to your comments, it was obvious once I've read them. I'm gonna go for a run before it's too dark, and come back with a cleaner head to finish this if you or the CI find any other issues :)